api/src/GearApi.ts

```diff
@@ -149,9 +148,33 @@ export class GearApi extends ApiPromise {
    * @param event
    * @returns
    */
-  getExtrinsicFailedError(event: Event): RegistryError {
+  getExtrinsicFailedError(event: Event): ExtrinsicFailedData {
     const error = event.data[0] as DispatchError;
-    const { isModule, asModule } = error;
-    return isModule ? this.registry.findMetaError(asModule) : null;
+    if (error.isModule) {
+      const data = this.registry.findMetaError(error.asModule);
+      return {
+        docs: data?.docs.join(' '),
+        method: data?.method,
+        name: data?.name,
+      };
+    } else if (error.isCannotLookup) {
```

Comment:
it seems that we can remove else here and keep only ifs. What do you think?

Reply:
Right! Thanks